pkg/tsdb/cloudwatch/credentials.go

```diff
@@ -60,6 +61,7 @@ func GetCredentials(dsInfo *DatasourceInfo) (*credentials.Credentials, error) {
 				&credentials.EnvProvider{},
 				&credentials.SharedCredentialsProvider{Filename: "", Profile: dsInfo.Profile},
 				remoteCredProvider(stsSess),
```

Comment:
@patstromm, could you set `webIdentityProvider` before `remoteCredProvider`? It's likely that the EC2 instances have a role attached to it, so this order will prioritise the EC2 instance role over the webIdentity. It's better to try first to use the WebIdentity